contracts/DxMgnPool.sol

```diff
@@ -80,9 +83,26 @@ contract DxMgnPool {
         uint auctionIndex = dx.getAuctionIndex(address(depositToken), address(secondaryToken));
         require(auctionIndex > lastParticipatedAuctionIndex, "Has to wait for new auction to start");
 
-        // ... super call into trader to participate in next auction
+        (address sellToken, address buyToken) =  assignBuyAndSellTokens();
+
+        uint depositAmount = depositToken.balanceOf(address(this));
+        if (isDepositTokenTurn && depositAmount > 0){
+            //depositng new tokens
+            require(depositToken.approve(address(dx), depositAmount), "Token transfer from Pool to dx was not approved");
+            dx.deposit(address(depositToken), depositAmount);
+        }
+
+        if (lastParticipatedAuctionIndex != 0)
```

Comment:
please use {} around if statements, even if they are just one line. Too easy to add another line later without noticing that we need the {}